src/components/CardEditor/CardEditForm/CardEditForm.jsx

```diff
@@ -27,67 +56,517 @@ const propTypes = {
+        </div>
+        <DataSeriesFormItem
+          value={value.content.series}
+          availableDataSourceIds={['temperature', 'humidity', 'pressure']}
+          onChange={val =>
+            onChange({
+              ...value,
+              content: {
+                ...(value.content ?? {}),
+                series: val,
+              },
+            })
+          }
+        />
+      </>
+    );
+  };
+
+  const renderTimeSeriesCardAdvancedItems = () => {
+    return (
+      <>
+        <div className={`${baseClassName}--input`}>
+          <TextInput
+            id="title"
+            labelText="X-axis label (Optional)"
+            light
+            onChange={evt =>
+              onChange({
+                ...value,
+                content: {
+                  ...(value.content ?? {}),
+                  xLabel: evt.target.value,
+                },
+              })
+            }
+            value={value.content?.xLabel}
+          />
+        </div>
+        <div className={`${baseClassName}--input`}>
+          <TextInput
+            id="title"
+            labelText="Y-axis label (Optional)"
+            light
+            onChange={evt =>
+              onChange({
+                ...value,
+                content: {
+                  ...(value.content ?? {}),
+                  yLabel: evt.target.value,
+                },
+              })
+            }
+            value={value.content?.yLabel}
+          />
+        </div>
+      </>
+    );
+  };
+  */
+
+  const commonFormItems = (
+    <>
+      <div className={`${baseClassName}--input`}>
+        <TextInput
+          id="title"
+          labelText="Card title"
+          light
+          onChange={evt => onChange({ ...value, title: evt.target.value })}
+          value={value.title}
+        />
+      </div>
+      <div className={`${baseClassName}--input`}>
+        <TextArea
+          id="description"
+          labelText="Description (Optional)"
+          light
+          onChange={evt => onChange({ ...value, description: evt.target.value })}
+          value={value.description}
+        />
+      </div>
+      <div className={`${baseClassName}--input`}>
+        <Dropdown
+          id="size"
+          label="Select a size"
+          direction="bottom"
+          itemToString={item => item.text}
+          items={(allowedSizesForType[value.type] ?? Object.keys(CARD_SIZES)).map(i => {
+            return {
+              id: i,
+              text: getCardSizeText(i),
+            };
+          })}
+          light
+          selectedItem={{ id: value.size, text: getCardSizeText(value.size) }}
+          onChange={({ selectedItem }) => {
+            onChange({ ...value, size: selectedItem.id });
+          }}
+          titleText="Size"
+        />
+      </div>
+    </>
+  );
+
   return (
     <>
+      {showEditor ? (
+        <CardCodeEditor
+          onSubmit={(val, setError) => {
+            try {
+              setError(false);
```

Comment:
It might be more efficient to set the error state at the end of the block as you're forcing a re-render before any of the logic has ran

Reply:
Good catch -- I'll address this in the next PR, as we'll want to replace this logic here anyway with a custom validation function from the parent.